pandas/tests/test_window.py

```diff
@@ -1130,14 +1130,41 @@ def test_rolling_quantile_np_percentile(self):
 
     def test_rolling_quantile_series(self):
         # #16211: Tests that rolling window's quantile default behavior
-        # is analogus to Series' quantile
+        # is analogous to Series' quantile
         arr = np.arange(100)
         s = Series(arr)
         q1 = s.quantile(0.1)
         q2 = s.rolling(100).quantile(0.1).iloc[-1]
 
         tm.assert_almost_equal(q1, q2)
 
+    @pytest.mark.parametrize('quantile', [0.0, 0.1, 0.45, 0.5, 1])
+    @pytest.mark.parametrize('na_probability', [0.0, 0.3])
+    @pytest.mark.parametrize('interpolation', ['linear', 'lower', 'higher',
+                                               'nearest', 'midpoint'])
+    def test_rolling_quantile_interpolation_options(self, quantile,
+                                                    na_probability,
+                                                    interpolation):
+        # Tests that rolling window's quantile behavior is analogous to
+        # Series' quantile for each interpolation option
+        size = 100
+        s = Series(np.random.rand(size))
```

Comment:
I'd advise against using random data in a test case as it could cause intermittent failures and make bugs harder to find. To inject missing data, could you not just parametrize the function with a second array of say `[0., np.nan, 0.2, np.nan, 0.4]` to match what you had earlier?